tests/integration/targets/ec2_asg/tasks/tag_operations.yml

```diff
@@ -0,0 +1,143 @@
+          - create_lc.failed is false
+          - '"autoscaling:CreateLaunchConfiguration" in create_lc.resource_actions'
+
+    #----------------------------------------------------------------------
+    - name: create a AutoScalingGroup to be used for tag_operations test
+      ec2_asg:
+        name: "{{ resource_prefix }}-asg-tag-test"
+        launch_config_name: "{{ resource_prefix }}-lc-tag-test"
+        health_check_period: 60
+        health_check_type: ELB
+        replace_all_instances: yes
+        min_size: 1
+        max_size: 1
+        desired_capacity: 1
+        region: "{{ aws_region }}"
+      register: create_asg
+
+    - name: ensure that AutoScalingGroup is created
+      assert:
+        that:
+          - create_asg is changed
+          - create_asg.failed is false
+          - '"autoscaling:CreateAutoScalingGroup" in create_asg.resource_actions'
+
+    #----------------------------------------------------------------------
+
+    - name: Get asg info
+      ec2_asg_info:
+        name: "{{ resource_prefix }}-asg-tag-test"
+      register: info_result
+
+    - debug: msg="{{ info_result }}"
+
+    - assert:
+        that:
+          - info_result.results[0].tags | length == 0
+
+    - name: Add 4 new tags - do not purge existing tags
+      ec2_asg:
+        name: "{{ resource_prefix }}-asg-tag-test"
+        tags:
+          - key1: 'val1'
+            propagate_at_launch: no
+          - key2: 'val2'
+            propagate_at_launch: yes
+          - key3: 'val3'
+            propagate_at_launch: yes
+          - key4: 'val4'
+            propagate_at_launch: no
+        purge_tags: false
+      register: add_result
+
+    - name: Get asg info
+      ec2_asg_info:
+        name: "{{ resource_prefix }}-asg-tag-test"
+      register: info_result
+
+    - assert:
+        that:
+          - add_result is changed
+          - info_result.results[0].tags | length == 4
+          - '"autoscaling:CreateOrUpdateTags" in add_result.resource_actions'
+
+    - name: Add 2 new tags - purge existing tags
+      ec2_asg:
+        name: "{{ resource_prefix }}-asg-tag-test"
+        tags:
+          - tag_a: 'val_a'
+            propagate_at_launch: no
+          - tag_b: 'val_b'
+            propagate_at_launch: yes
+      register: add_purge_result
+
+    - name: Get asg info
+      ec2_asg_info:
+        name: "{{ resource_prefix }}-asg-tag-test"
+      register: info_result
+
+    - assert:
+        that:
+          - add_purge_result is changed
+          - info_result.results[0].tags | length == 2
+          - '"autoscaling:CreateOrUpdateTags" in add_purge_result.resource_actions'
+
+    - name: Add 2 more tags - do not purge existing tags
+      ec2_asg:
+        name: "{{ resource_prefix }}-asg-tag-test"
+        tags:
+          - key1: 'val1'
+            propagate_at_launch: no
+          - key2: 'val2'
+            propagate_at_launch: yes
+        purge_tags: false
+      register: add_result
+
```

Comment:
Can you also add another task with `tags: []` with  `purge_tags: false` and check if the existing tags are retained?